content/docs/latest/installing/vms/hyper-v.md

```diff
@@ -0,0 +1,153 @@
+# create the config drive
+& "mkisofs.exe" -o "config-drive.iso" -ignore-error -ldots -allow-lowercase -allow-multidot -l -publisher "cbsl" -quiet -J -r -V "config-2" "config-drive-metadata"
+```
+
+An ISO file named `config-drive.iso` will be created that will configure a virtual machine to accept your SSH key and set its name to my_flatcar_01.
+
+## Deploying a new virtual machine on Hyper-V using config drive
+
+```powershell
+$vmName = "my_flatcar_01"
+$vmDisk = "flatcar_production_hyperv_vhdx_image.vhdx"
+
+# For Generation 1 VM
+New-VM -Name $vmName -MemoryStartupBytes 2GB `
+    -BootDevice VHD -SwitchName "Default Switch" -VHDPath $vmDisk -Generation 1
+
+# For Generation 2 VM
+New-VM -Name $vmName -MemoryStartupBytes 2GB `
+    -BootDevice VHD -SwitchName "Default Switch" -VHDPath $vmDisk -Generation 2
+# Generation 2 VM needs to have secure boot disabled, as the images are not signed
+Set-VMFirmware -EnableSecureBoot "Off" -VMName $vmName
+
+# Now, add the config-drive file as DvdDrive
+Add-VMDvdDrive -VMName $vmName -Path "config-drive.iso"
+
+Start-VM -Name $vmName
+```
+
+## Advanced example: deploying a new virtual machine on Hyper-V using Ignition with autologin and TPM LUKS2 root partition encryption
+
+```powershell
+$vmName = "my_flatcar_01"
+$vmDisk = "flatcar_production_hyperv_vhdx_image.vhdx"
+
+New-VM -Name $vmName -MemoryStartupBytes 2GB `
+    -BootDevice VHD -SwitchName "Default Switch" -VHDPath $vmDisk -Generation 2
+Set-VMFirmware -EnableSecureBoot "Off" -VMName $vmName
+
+# The core user password is set to foo
+
+$ignitionMetadata = @'
+variant: flatcar
+version: 1.0.0
+kernel_arguments:
+  should_exist:
+    - flatcar.autologin
+passwd:
+  users:
+    - name: core
+      password_hash: $6$sn3ZSJJJln5JkAZb$VDTKzLpCyjlEe7Kh0DKjOnEawkkOoi0tOKVbcCv0FIWSf3u9Y1p1I5YdJJ5L8uDmmMvO2CBlmJZNdxFuekjjE1
+storage:
+  luks:
+  - name: rootencrypted
+    wipe_volume: true
+    device: "/dev/disk/by-partlabel/ROOT"
+  filesystems:
+    - device: /dev/mapper/rootencrypted
+      format: ext4
+      label: ROOT
+systemd:
+  units:
+    - name: cryptenroll-helper.service
+      enabled: true
+      contents: |
+        [Unit]
+        ConditionFirstBoot=true
+        OnFailure=emergency.target
+        OnFailureJobMode=isolate
+        [Service]
+        Type=oneshot
+        RemainAfterExit=yes
+        ExecStart=systemd-cryptenroll --tpm2-device=auto --unlock-key-file=/etc/luks/rootencrypted --wipe-slot=0 --tpm2-pcrs= /dev/disk/by-partlabel/ROOT
+        ExecStart=rm /etc/luks/rootencrypted
+        [Install]
+        WantedBy=multi-user.target
+'@
+echo $ignitionMetadata > ignition.yaml
+
+# download the butane binary to create the raw ignition metadata
+# https://github.com/coreos/butane/releases
+
+# transform the Ignition metadata from Butane format to Ignition raw
+cmd /c "butane-x86_64-pc-windows-gnu.exe < ignition.yaml > ignition.json"
```

Comment:
why "cmd /c" at all?

Reply:
binaries like butane do not work well with PowerShell as the `<` is seen as a comparator and not as a redirection. cmd.exe is the only way to make sure the construct `< >` works similar to bash.

```
The '<' operator is reserved for future use.
```

A working way in PowerShell would be:

```
cat .\ignition.yaml | & "butane-x86_64-pc-windows-gnu.exe" > .\ignition.json
```

Reply:
then i would probably prefer to call it as:
```
butane-x86_64-pc-windows-gnu.exe .\ignition.yaml -o .\ignition.json
```